app/components/supplementary-file-browser/component.js

```diff
@@ -0,0 +1,132 @@
+import Ember from 'ember';
+import loadAll from 'ember-osf/utils/load-relationship';
+import config from 'ember-get-config';
+
+function fileDownloadPath(file, node) {
+    if (!file || !node) {
+        return;
+    }
+    if (file.get('guid')) {
+        return `${config.OSF.url}${file.get('guid')}/?action=download`;
+    }
+    return `${config.OSF.url}project/${node.get('id')}/files/osfstorage${file.get('path')}/?action=download`;
+}
+
+/**
+ * @module ember-osf-reviews
+ * @submodule components
+ */
+
+/**
+ * Displays supplemental preprint files
+ *
+ * Sample usage:
+ * ```handlebars
+ * {{supplementary-file-browser
+ *      preprint=model
+ *      node=node
+ *      projectURL=node.links.html
+ *      chooseFile=(action 'chooseFile')
+ * }}
+ * ```
+ * @class supplementary-file-browser
+ */
+export default Ember.Component.extend({
+    elementId: 'preprint-file-view',
+    endIndex: 6,
+    startIndex: 0,
+
+    scrollAnim: '',
+    selectedFile: null,
+
+    hasAdditionalFiles: function() {
```

Comment:
I can't believe I'm saying this but usage of the function prototype extensions is frowned upon. Prefer `Ember.computed`.